From a2e0f578d1ae6c2d42e8513851de101329985901 Mon Sep 17 00:00:00 2001
From: Jeff Layton <jlayton@redhat.com>
Date: Mon, 21 Apr 2014 12:17:57 -0400
Subject: [fs] nfs: clean up, revert increase in READDIR RPC buffer max size

Message-id: <1398082678-3620-4-git-send-email-jlayton@redhat.com>
Patchwork-id: 78977
O-Subject: [RHEL7.1 PATCH 3/4] [BZ#1048477] NFS: Clean up: revert increase in READDIR RPC buffer max size
Bugzilla: 1048477
RH-Acked-by: Steve Dickson <SteveD@redhat.com>
RH-Acked-by: J. Bruce Fields <bfields@redhat.com>

From: Chuck Lever <chuck.lever@oracle.com>

Security labels go with each directory entry, thus they are always
stored in the page cache, not in the head buffer.  The length of the
reply that goes in head[0] should not have changed to support
NFSv4.2 labels.

Signed-off-by: Chuck Lever <chuck.lever@oracle.com>
Signed-off-by: Trond Myklebust <trond.myklebust@primarydata.com>
Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1048477
Upstream: a7697f6ff8e853d5cf443ad60445b99114b15575
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfs/nfs4xdr.c b/fs/nfs/nfs4xdr.c
index 72f3bf1..73ce8d4 100644
--- a/fs/nfs/nfs4xdr.c
+++ b/fs/nfs/nfs4xdr.c
@@ -203,8 +203,7 @@ static int nfs4_stat_to_errno(int);
      2 + encode_verifier_maxsz + 5 + \
     nfs4_label_maxsz)
 #define decode_readdir_maxsz (op_decode_hdr_maxsz + \
-     decode_verifier_maxsz + \
-    nfs4_label_maxsz + nfs4_fattr_maxsz)
+     decode_verifier_maxsz)
 #define encode_readlink_maxsz (op_encode_hdr_maxsz)
 #define decode_readlink_maxsz (op_decode_hdr_maxsz + 1)
 #define encode_write_maxsz (op_encode_hdr_maxsz + \
-- 
1.7.1